### ISSUE TITLE
Elements and Tags: <Short description of your suggestion> Malware detection notification

### ISSUE THREAD

**Issue by codiaks** [TheOdinProject/curriculum#28151]
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [x] Would you like to work on this issue?

### Describe your suggestion

![image](https://github.com/TheOdinProject/curriculum/assets/54658593/1aac9c86-0213-4869-ba05-7516743e82e6)
The redirect link was not opening and showed a notification in chrome saying that a malware was detected.
Why ???
Now I am afraid to go through any other links and pages by you guys...

### Path

Foundations

### Lesson Url

https://www.theodinproject.com/lessons/foundations-elements-and-tags

**Comment by CoderModer1** [comment]
Looks like you didn't pay attention to the foundation lessons:

HTTP (Hypertext Transfer Protocol) is an [application layer](https://en.wikipedia.org/wiki/Application_layer) protocol in the [Internet protocol suite](https://en.wikipedia.org/wiki/Internet_protocol_suite) model for distributed, collaborative, [hypermedia](https://en.wikipedia.org/wiki/Hypermedia) information systems.[[1]](https://en.wikipedia.org/wiki/HTTP#cite_note-rfc9110-1) HTTP is the foundation of data communication for the [World Wide Web](https://en.wikipedia.org/wiki/World_Wide_Web), where [hypertext](https://en.wikipedia.org/wiki/Hypertext) documents include [hyperlinks](https://en.wikipedia.org/wiki/Hyperlink) to other resources that the user can easily access, for example by a [mouse](https://en.wikipedia.org/wiki/Computer_mouse) click or by tapping the screen in a web browser.

Development of HTTP was initiated by [Tim Berners-Lee](https://en.wikipedia.org/wiki/Tim_Berners-Lee) at [CERN](https://en.wikipedia.org/wiki/CERN) in 1989 and summarized in a simple document describing the behavior of a client and a server using the first HTTP version, named 0.9.[[2]](https://en.wikipedia.org/wiki/HTTP#cite_note-HTTP/0.9-specifications-2) That version was subsequently developed, eventually becoming the public 1.0.[[3]](https://en.wikipedia.org/wiki/HTTP#cite_note-HTTP/1.0-first-unofficial-draft-3)

Development of early HTTP [Requests for Comments](https://en.wikipedia.org/wiki/Requests_for_Comments) (RFCs) started a few years later in a coordinated effort by the [Internet Engineering Task Force](https://en.wikipedia.org/wiki/Internet_Engineering_Task_Force) (IETF) and the [World Wide Web Consortium](https://en.wikipedia.org/wiki/World_Wide_Web_Consortium) (W3C), with work later moving to the IETF.

HTTP/1 was finalized and fully documented (as version 1.0) in 1996.[[4]](https://en.wikipedia.org/wiki/HTTP#cite_note-4) It evolved (as version 1.1) in 1997 and then its specifications were updated in 1999, 2014, and 2022.[[5]](https://en.wikipedia.org/wiki/HTTP#cite_note-5)

Its secure variant named [HTTPS](https://en.wikipedia.org/wiki/HTTPS) is used by more than 85% of websites.[[6]](https://en.wikipedia.org/wiki/HTTP#cite_note-HTTPS-usage-web-servers-6) [HTTP/2](https://en.wikipedia.org/wiki/HTTP/2), published in 2015, provides a more efficient expression of HTTP's semantics "on the wire". As of January 2024, it is used by 36% of websites[[7]](https://en.wikipedia.org/wiki/HTTP#cite_note-HTTP2-usage-web-servers-7) and supported by almost all web browsers (over 98% of users).[[8]](https://en.wikipedia.org/wiki/HTTP#cite_note-HTTP2-Can-I-Use-8) It is also supported by major web servers over [Transport Layer Security](https://en.wikipedia.org/wiki/Transport_Layer_Security) (TLS) using an [Application-Layer Protocol Negotiation](https://en.wikipedia.org/wiki/Application-Layer_Protocol_Negotiation) (ALPN) extension[[9]](https://en.wikipedia.org/wiki/HTTP#cite_note-rfc7301-9) where [TLS 1.2](https://en.wikipedia.org/wiki/TLS_1.2) or newer is required.[[10]](https://en.wikipedia.org/wiki/HTTP#cite_note-10)[[11]](https://en.wikipedia.org/wiki/HTTP#cite_note-11)

[HTTP/3](https://en.wikipedia.org/wiki/HTTP/3), the successor to HTTP/2, was published in 2022.[[12]](https://en.wikipedia.org/wiki/HTTP#cite_note-12) As of February 2024, it is now used on 29% of websites[[13]](https://en.wikipedia.org/wiki/HTTP#cite_note-HTTP3-usage-web-servers-13) and is supported by most web browsers, i.e. (at least partially) supported by 97% of users.[[14]](https://en.wikipedia.org/wiki/HTTP#cite_note-HTTP3-Can-I-Use-14) HTTP/3 uses [QUIC](https://en.wikipedia.org/wiki/QUIC) instead of [TCP](https://en.wikipedia.org/wiki/Transmission_Control_Protocol) for the underlying transport protocol. Like HTTP/2, it does not obsolesce previous major versions of the protocol. Support for HTTP/3 was added to [Cloudflare](https://en.wikipedia.org/wiki/Cloudflare) and [Google Chrome](https://en.wikipedia.org/wiki/Google_Chrome) first,[[15]](https://en.wikipedia.org/wiki/HTTP#cite_note-15)[[16]](https://en.wikipedia.org/wiki/HTTP#cite_note-16) and is also enabled in [Firefox](https://en.wikipedia.org/wiki/Firefox).[[17]](https://en.wikipedia.org/wiki/HTTP#cite_note-17) HTTP/3 has lower latency for real-world web pages, if enabled on the server, and loads faster than with HTTP/2, in some cases over three times faster than HTTP/1.1 (which is still commonly only enabled).[[18]](https://en.wikipedia.org/wiki/HTTP#cite_note-18)

The reason why an browser may be giving an wearing is that its using http instead of https that is its not encrypted therefore its considered not secure, I hope you find peace in your life.

PS: Don't show of with issues, go suck some tits

**Comment by CoderModer1** [comment]
Looks like you didn't pay attention to the foundation lessons:

HTTP (Hypertext Transfer Protocol) is an [application layer](https://en.wikipedia.org/wiki/Application_layer) protocol in the [Internet protocol suite](https://en.wikipedia.org/wiki/Internet_protocol_suite) model for distributed, collaborative, [hypermedia](https://en.wikipedia.org/wiki/Hypermedia) information systems.[[1]](https://en.wikipedia.org/wiki/HTTP#cite_note-rfc9110-1) HTTP is the foundation of data communication for the [World Wide Web](https://en.wikipedia.org/wiki/World_Wide_Web), where [hypertext](https://en.wikipedia.org/wiki/Hypertext) documents include [hyperlinks](https://en.wikipedia.org/wiki/Hyperlink) to other resources that the user can easily access, for example by a [mouse](https://en.wikipedia.org/wiki/Computer_mouse) click or by tapping the screen in a web browser.

Development of HTTP was initiated by [Tim Berners-Lee](https://en.wikipedia.org/wiki/Tim_Berners-Lee) at [CERN](https://en.wikipedia.org/wiki/CERN) in 1989 and summarized in a simple document describing the behavior of a client and a server using the first HTTP version, named 0.9.[[2]](https://en.wikipedia.org/wiki/HTTP#cite_note-HTTP/0.9-specifications-2) That version was subsequently developed, eventually becoming the public 1.0.[[3]](https://en.wikipedia.org/wiki/HTTP#cite_note-HTTP/1.0-first-unofficial-draft-3)

Development of early HTTP [Requests for Comments](https://en.wikipedia.org/wiki/Requests_for_Comments) (RFCs) started a few years later in a coordinated effort by the [Internet Engineering Task Force](https://en.wikipedia.org/wiki/Internet_Engineering_Task_Force) (IETF) and the [World Wide Web Consortium](https://en.wikipedia.org/wiki/World_Wide_Web_Consortium) (W3C), with work later moving to the IETF.

HTTP/1 was finalized and fully documented (as version 1.0) in 1996.[[4]](https://en.wikipedia.org/wiki/HTTP#cite_note-4) It evolved (as version 1.1) in 1997 and then its specifications were updated in 1999, 2014, and 2022.[[5]](https://en.wikipedia.org/wiki/HTTP#cite_note-5)

Its secure variant named [HTTPS](https://en.wikipedia.org/wiki/HTTPS) is used by more than 85% of websites.[[6]](https://en.wikipedia.org/wiki/HTTP#cite_note-HTTPS-usage-web-servers-6) [HTTP/2](https://en.wikipedia.org/wiki/HTTP/2), published in 2015, provides a more efficient expression of HTTP's semantics "on the wire". As of January 2024, it is used by 36% of websites[[7]](https://en.wikipedia.org/wiki/HTTP#cite_note-HTTP2-usage-web-servers-7) and supported by almost all web browsers (over 98% of users).[[8]](https://en.wikipedia.org/wiki/HTTP#cite_note-HTTP2-Can-I-Use-8) It is also supported by major web servers over [Transport Layer Security](https://en.wikipedia.org/wiki/Transport_Layer_Security) (TLS) using an [Application-Layer Protocol Negotiation](https://en.wikipedia.org/wiki/Application-Layer_Protocol_Negotiation) (ALPN) extension[[9]](https://en.wikipedia.org/wiki/HTTP#cite_note-rfc7301-9) where [TLS 1.2](https://en.wikipedia.org/wiki/TLS_1.2) or newer is required.[[10]](https://en.wikipedia.org/wiki/HTTP#cite_note-10)[[11]](https://en.wikipedia.org/wiki/HTTP#cite_note-11)

[HTTP/3](https://en.wikipedia.org/wiki/HTTP/3), the successor to HTTP/2, was published in 2022.[[12]](https://en.wikipedia.org/wiki/HTTP#cite_note-12) As of February 2024, it is now used on 29% of websites[[13]](https://en.wikipedia.org/wiki/HTTP#cite_note-HTTP3-usage-web-servers-13) and is supported by most web browsers, i.e. (at least partially) supported by 97% of users.[[14]](https://en.wikipedia.org/wiki/HTTP#cite_note-HTTP3-Can-I-Use-14) HTTP/3 uses [QUIC](https://en.wikipedia.org/wiki/QUIC) instead of [TCP](https://en.wikipedia.org/wiki/Transmission_Control_Protocol) for the underlying transport protocol. Like HTTP/2, it does not obsolesce previous major versions of the protocol. Support for HTTP/3 was added to [Cloudflare](https://en.wikipedia.org/wiki/Cloudflare) and [Google Chrome](https://en.wikipedia.org/wiki/Google_Chrome) first,[[15]](https://en.wikipedia.org/wiki/HTTP#cite_note-15)[[16]](https://en.wikipedia.org/wiki/HTTP#cite_note-16) and is also enabled in [Firefox](https://en.wikipedia.org/wiki/Firefox).[[17]](https://en.wikipedia.org/wiki/HTTP#cite_note-17) HTTP/3 has lower latency for real-world web pages, if enabled on the server, and loads faster than with HTTP/2, in some cases over three times faster than HTTP/1.1 (which is still commonly only enabled).[[18]](https://en.wikipedia.org/wiki/HTTP#cite_note-18)

The reason why a browser may be giving a warning is that its using http instead of https i.e its not encrypted therefore its considered not secure, I hope you find peace in your life.

PS: Don't show off with issues, go suck some tits.

**Comment by sukairaida** [comment]
@CoderModer1 Did you just create your account to be offensive and try to humilliate this person? People from all different backgrounds follow the curriculum and them being afraid of a warning from their browser is a valid concern. Just copy-pasting something and then treating them poorly does not help them or anyone at all.

**Comment by MaoShizhong** [comment]
@CoderModer1, that is absolutely not an appropriate response to someone raising a genuine concern.
We have our [community rules and expectations](https://www.theodinproject.com/guides/community/rules) which also apply to GitHub. The way you have responded is straight up rude and not helpful at all.

@codiaks Please ignore the way you were responded to in the above message. What you raised is a perfectly valid concern, especially if you're not aware of the cause.

The reason you got a security notification was because the website uses `http` instead of `https`, where `http` is not secure, and some browsers/settings will give a strong warning against `http` sites due to the lack of security.

While most sites will auto-redirect you to the `https` version if the link only has `http` in it, this website doesn't seem to have an `https` version.

What we should do is find an alternative resource that provides the same points, but is accessible via `https`, then replace the current link with the new one. If you can find an appropriate replacement resource, then I can assign this issue to you :)

**Comment by codiaks** [comment]
@MaoShizhong,
The issue I faced was not only an HTTP not-secure warning but also for a moment a notification was shown saying malware was detected.

I searched online to see if there were any reported issues with the Odin Project, but I couldn't find anything. I then tried using the contact button on the Odin website, but it redirected me to GitHub. I searched for any reported issues on GitHub but didn't find any. Ultimately, I decided to create a new issue. In hindsight, I realize I should have asked for help on Discord instead. That's the story of what happened. I tested the link on a mobile device and didn't encounter any issues. 

After reviewing the content again, I believe the current link is quite good and people may continue to find it useful. Therefore, I suggest not replacing it at the moment. However, if more people report issues with the link, we can consider replacing it with something else. This is my perspective on the matter. If you still believe it should be replaced, I can look for an alternative resource.

**Comment by MaoShizhong** [comment]
@codiaks the malware warning you would've gotten in this case likely would've been your browser being over-protective.

We should still replace the link with a similar resource that can be accessed via HTTPS.
Most sites will auto-redirect to the HTTPS version when you hse the HTTP link, but this resource doesn't for some reason, only having an HTTP version. Therefore, we should find an HTTPS replacement for it.

**Comment by MaoShizhong** [comment]
We can just replace the href for that link with the link to the video itself (which uses HTTPS) instead of the page that houses the video, since the page doesn't really include much necessary stuff outside of the video.

The direct URL for the video is `https://player.vimeo.com/video/24549728`

@codiaks would you be happy to replace that link's href with the above vimeo URL instead? If so, I can assign this issue to you, then you can open a PR once assigned. If not, let me know so I can open this issue up for assignment.

**Comment by MaoShizhong** [comment]
Opening this up for contribution.

## Acceptance criteria

- [ ] The href for the link in Additional resources is replaced with `https://player.vimeo.com/video/24549728`. No other wording needs changing.

Please comment below if you'd like to be assigned to this issue. Please do not open a PR unless you have been assigned by a maintainer.

**Comment by ItsTosh** [comment]
Hey 👋 I’d be happy to contribute

**Comment by MaoShizhong** [comment]
All yours

**Comment by ItsTosh** [comment]
Sorry it took long, been busy all day!
